locations/spiders/averitt.py

```diff
@@ -1,69 +1,118 @@
+                "string(/html/body/div[2]/div/div[1]/div/div/span/div[4]/div/div/div/div[2]/div/div/p)"
+            ).get()
+            if address_full:
+                city_state_postcode = address.replace(address_full, "")
+                postcode = city_state_postcode.split()[-1]
+                state = city_state_postcode.split()[-2]
+                city = " ".join(city_state_postcode.split()[:-2]).replace(",", "")
 
+        # 4
+        if city == None:
+            address_full = response.xpath(
+                "/html/body/div[2]/div/div[1]/div/div/span/div[4]/div/div/div/div[2]/div/div/p[1]/span/text()"
+            ).get()
+            address = response.xpath(
+                "string(/html/body/div[2]/div/div[1]/div/div/span/div[4]/div/div/div/div[2]/div/div/p[1]/span)"
+            ).get()
+            if address_full:
+                city_state_postcode = address.replace(address_full, "")
+                postcode = city_state_postcode.split()[-1]
+                state = city_state_postcode.split()[-2]
+                city = " ".join(city_state_postcode.split()[:-2]).replace(",", "")
+
+        ref_map_url = (
+            response.xpath(
+                '//a[contains(@href,"goo.gl") or contains(@href, "g.page")]/@href'
+            )
+            .get()
+            .replace("https://goo.gl/maps/", "")
+            .replace("https://g.page/", "")
+        )
+        phone = response.xpath(
+            "/html/body/div[2]/div/div[1]/div/div/span/div[2]/div/div/div/div[1]/div/div/p/a/text()"
+        ).get()
+        email = (
+            response.xpath(
+                "/html/body/div[2]/div/div[1]/div/div/span/div[2]/div/div/div/div[2]/div/div/span/a/@href"
+            )
+            .get()
+            .replace("mailto:", "")
+        )
+        name = response.xpath(
+            "/html/body/div[2]/div/div[1]/div/div/span/div[4]/div/div/div/div[2]/div/div/h3/strong/text()"
+        ).get()
         properties = {
-            "ref": "_".join(
-                re.search(r".+/(.+?)/(.+?)/?(?:\.jsp|$)", response.url).groups()
-            ),
-            "name": response.xpath(
-                '//section[contains(@class, "content-main")]//h4/text()'
-            ).extract_first(),
-            "addr_full": response.xpath(
-                'normalize-space(//span[@class="street-address"]//text())'
-            ).extract_first(),
-            "city": response.xpath(
-                'normalize-space(//span[@class="locality"]//text())'
-            ).extract_first(),
-            "state": response.xpath(
-                'normalize-space(//span[@class="region"]//text())'
-            ).extract_first(),
-            "postcode": response.xpath(
-                'normalize-space(//span[@class="postal-code"]//text())'
-            ).extract_first(),
-            "website": response.url,
-            "lat": float(lat) if lat else None,
-            "lon": float(lon) if lon else None,
+            "ref": "_" + ref_map_url,
```

Comment:
I'm not in love with using this Google id as our ref. I doubt there is any real issue with it, but we could just use `response.url`

Reply:
Also, we could potentially use `extract_google_position` to get the coords. It's not perfect but it may work on this site.